File: src/main/frontend/src/components/edit-action-component.ts
import {html, LitElement} from "lit";
import {customElement} from "lit/decorators.js";
import '@vaadin/button';
import styles from "./edit-action-component.css?inline";

@customElement("edit-action-component")
class EditActionComponent extends LitElement {

    static get styles() {
        return [styles]
    }

    render() {
        return html`
            <div class="edit-action-panel">
                <vaadin-button id="saveButton">Сохранить</vaadin-button>
                <vaadin-button id="cancelButton">Отмена</vaadin-button>
            </div>
        `;
    }

}

File: src/main/java/ru/menshevva/demoapp/ui/components/EditActionComponent.java
package ru.menshevva.demoapp.ui.components;

import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.littemplate.LitTemplate;
import com.vaadin.flow.component.template.Id;

@Tag("edit-action-component")
@JsModule("./src/components/edit-action-component.ts")
public class EditActionComponent extends LitTemplate {

    @Id
    private Button saveButton;

    @Id
    private Button cancelButton;

    public EditActionComponent(EditActionCallback editActionCallback) {
        if (editActionCallback != null) {
            saveButton.addClickListener(event -> {
                editActionCallback.ok();
            });
            cancelButton.addClickListener(event -> {
                editActionCallback.cancel();
            });
        }
    }


}
